### PR TITLE
Remove cfg-if

### LINKED ISSUE #225: Consider replacing cfg_if with ordinary cfg attributes
**Is your feature request related to a problem? Please describe.**

I have seen `cfg_if!` used to great effect, [for example here in libc](https://github.com/rust-lang/libc/blob/0.2.40/src/unix/mod.rs#L271) where it is a huge maintainability improvement over using plain `#[cfg]`.

Skimming lib.rs of uuid I didn't really see the benefit. Here are some extracts to illustrate what I mean.

**Describe the solution you'd like**

---

##### Before

```rust
// I find it hard to parse out from this the fact that we depend on slog if and
// only if `feature = "slog"` is enabled.
cfg_if! {
    if #[cfg(all(feature = "slog", not(test)))] {
        extern crate slog;
    } else if #[cfg(all(feature = "slog", test))] {
        #[macro_use]
        extern crate slog;
    }
}
```

##### After

```rust
#[cfg(feature = "slog")]
#[cfg_attr(test, macro_use)]
extern crate slog;
```

---

##### Before

```rust
// What's the point?
cfg_if! {
    if #[cfg(feature = "serde")] {
        mod serde_support;
    }
}
```

##### After

```rust
#[cfg(feature = "serde")]
mod serde_support;
```

---

##### Before

```rust
// This is a lot of code, punctuation, and indentation (4 levels!) to set up a
// consistent import hierarchy. Since only core types are imported, the only
// relevant difference between std and no-std is that the no-std prelude does
// not include the core crate.
cfg_if! {
    if #[cfg(feature = "std")] {
        use std::fmt;
        use std::str;

        cfg_if! {
            if #[cfg(feature = "v1")] {
                use std::sync::atomic;
            }
        }
    } else if #[cfg(not(feature = "std"))] {
        use core::fmt;
        use core::str;

        cfg_if! {
            if #[cfg(feature = "v1")] {
                use core::sync::atomic;
            }
        }
    }
}
```

##### After

```rust
#[cfg(feature = "std")]
extern crate core;

use core::{fmt, str};

#[cfg(feature = "v1")]
use core::sync::atomic;
```

---

**Is it blocking?**

This issue is not blocking any of my work but may present a hurdle to new uuid contributors.

**Additional context**

Previous discussion of `cfg_if!` in https://github.com/uuid-rs/uuid/pull/167.

### PR DESCRIPTION
**Pull request by kinggoesgaming**
**I'm submitting a ...**
  - [ ] bug fix
  - [ ] feature enhancement
  - [x] deprecation or removal
  - [x] refactor

# Description
Remove `cfg-if` dependency from the project

# Motivation
Originally the `cfg-if` dependency was introduced for help easy reading of code, as it was mostly in one file. However with the various refactor attempts, as described in #225, it has just ended up creating only noise.

# Tests
No new tests needed, current tests pass

# Related Issue(s)
closes #225

### CODE REVIEW

File: src/v1.rs [hunk 1]
```diff
@@ -161,7 +161,8 @@ mod tests {
         let context = UuidContext::new(0);
 
         {
-            let uuid = Uuid::new_v1(&context, time, time_fraction, &node).unwrap();
+            let uuid =
```
Comment (Dylan-DPC) [inline src/v1.rs]: is this from latest fmt? looks messed up

Comment (kinggoesgaming) [reply src/v1.rs]: yes its the latest nightly fmt...

File: src/lib.rs [hunk 2]
```diff
@@ -117,115 +117,45 @@
...
-}
+#[cfg(feature = "md5")]
+extern crate md5;
+#[cfg(feature = "rand")]
+extern crate rand;
+#[cfg(feature = "serde")]
+extern crate serde;
+#[cfg(feature = "sha1")]
+extern crate sha1;
+#[cfg(feature = "slog")]
```
[the remark this answers is not in the document]
Comment (kinggoesgaming) [reply src/lib.rs]: er no it does not seem like we are... but thats for another PR

[the remark this answers is not in the document]
Comment (kinggoesgaming) [reply src/lib.rs]: actually appveyor run does test it, but travis

**Comment by dtolnay** [comment]
Thanks for the quick fix. dbe76ede201a10c4bd7f672284a9200f4a27f7e5 is a huge improvement.

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by kinggoesgaming** [comment]
Rustfmt settings changed :(

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by kinggoesgaming** [comment]
bors ping
